app/models/division.rb

```diff
@@ -50,8 +50,7 @@ def self.root_internal_name
 
   def self.root
     # create on demand if not present for convenience of blank db's and test cases
-    # @@root ||=
```

Comment:
The main challenge with the root division is not code complexity, but simply the endless discussions around it.  The majority of the dependencies are around building the unit test and migration data.  Beyond access control, the division hierarchy is key to resolving which option sets and custom fields to use for a particular entity since these are potentially customized per division.

Anyways, the latest should reflect all of Mel's recommendations, and I've only logged a small fraction of my own that has gone into the code review discussions and updates.  

A couple of notes, the root division primary key was a quick hack when I first built out the prototype flows, and had been removed quite a while back.  That had been replaced with a special 'internal_name', but I've now removed that and the code assumes there is a unique root node without a parent.  The current live code does not verify that Division.root returns a value.  This results in obscure errors in the unit test execution when the root division is not properly initialized.  I have a version of the Division.root logic implemented, but commented out, which was very useful in getting the unit tests working again once the logic to autocreate the root division was removed.  I'm still partial to re-enabling the version of Division.root which throws a helpful exception if the root division is missing.